src/extension/util.rs

```diff
@@ -38,12 +38,15 @@ where
     let name = str::from_utf8(name)?;
     let ext = parse_extension_element(reader, atts)?;
 
-    let map = extensions
-        .entry(ns.to_string())
-        .or_insert_with(BTreeMap::new);
-
-    let items = map.entry(name.to_string()).or_insert_with(Vec::new);
-    items.push(ext);
+    if let Some(map) = extensions.get_mut(ns) {
+        if let Some(items) = map.get_mut(name) {
+            items.push(ext);
+        } else {
+            map.insert(name.to_string(), vec![ext]);
+        }
+    } else {
+        extensions.insert(ns.to_string(), BTreeMap::new());
```

Comment:
Changed code does not do the same. There is no insertion of `name`->`ext` pair in this branch.